test/distributed/test_c10d_nccl.py

```diff
@@ -2626,6 +2626,11 @@ def test_nccl_barrier_device_ids_function_argument(self):
         with self.assertRaisesRegex(RuntimeError, "Invalid function argument"):
             c10d.barrier(device_ids=self.rank)
 
+    @requires_nccl()
+    @skip_if_lt_x_gpu(2)
+    def test_nccl_warn_not_in_group(self):
```

Comment:
should we also test it works fine with TORCH_DISTRIBUTED_DEBUG? It should because this is in the python level, but it would be good to make sure the wrapped process group raises this error as expected. 

You can use `with_dist_debug` decorator to do it - https://github.com/pytorch/pytorch/blob/master/test/distributed/test_c10d_nccl.py#L1068

Reply:
Yep, let me add a test.